hack/capi-sync-provider.sh

```
@@ -0,0 +1,117 @@
+            "$VENDOR_DIR/vendor/sigs.k8s.io/cluster-api-provider-aws/v2/exp/api/v1beta1" \
+            "$DST_DIR/exp/v1beta1" \
+            --allow sigs.k8s.io/cluster-api-provider-aws/v2/api/ \
+            --rewrite "sigs.k8s.io/cluster-api-provider-aws/v2/api/=github.com/openshift/hypershift/pkg/capi/aws/"
+        copy_types \
+            "$VENDOR_DIR/vendor/sigs.k8s.io/cluster-api-provider-aws/v2/exp/api/v1beta2" \
+            "$DST_DIR/exp/v1beta2" \
+            --allow sigs.k8s.io/cluster-api-provider-aws/v2/api/ \
+            --rewrite "sigs.k8s.io/cluster-api-provider-aws/v2/api/=github.com/openshift/hypershift/pkg/capi/aws/"
+        # Remove ROSA/EKS/Fargate/Managed types — HyperShift doesn't use them and
+        # stripping leaves broken cross-references to banned rosacontrolplane imports.
+        find "$DST_DIR" \( -name 'rosa*' -o -name 'awsfargate*' -o -name 'awsmanaged*' \) -print | xargs rm -f
```

Shellcheck (0.11.0)

[warning] Use 'find .. -print0 | xargs -0 ..' or 'find .. -exec .. +' to allow non-alphanumeric filenames.

(SC2038)